test-support/src/test/java/org/springframework/ldap/test/unboundid/EmbeddedLdapServerTests.java

```diff
@@ -19,85 +19,161 @@
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import javax.naming.NamingException;
+import javax.naming.directory.Attributes;
 
 import com.unboundid.ldap.listener.InMemoryDirectoryServer;
 import com.unboundid.ldap.listener.InMemoryDirectoryServerConfig;
 import com.unboundid.ldap.listener.InMemoryListenerConfig;
+import org.junit.Before;
 import org.junit.Test;
 
+import org.springframework.ldap.core.AttributesMapper;
+import org.springframework.ldap.core.LdapTemplate;
+import org.springframework.ldap.core.support.LdapContextSource;
+import org.springframework.ldap.query.LdapQueryBuilder;
+
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
 
 public class EmbeddedLdapServerTests {
 
+	private int port;
+
+	@Before
+	public void setUp() {
+		this.port = getFreePort();
+	}
+
 	@Test
-	public void shouldStartAndCloseServer() throws Exception {
```

Comment:
Is there a reason this test needs to be removed? It would be best to leave existing tests as-is to ensure we haven't broken anything.

Reply:
Since `EmbeddedLdapServer::newEmbeddedServer` now also uses the builder, we had some overlapping tests which I tried to remove/merge.

However, the coverage was not impacted: pretty much everything is covered by tests (apart from 2-3 ldap exceptions)

Reply:
Okay, makes sense. To make it clearer what was needed to add this feature, will you please move the test polishing you did to an earlier commit? 

The PR is ready when the feature commit only has the things that are needed to add that feature and since these changes aren't needed (though they are most welcome!), they should happen in a separate commit. You could add a commit like:

```
Polish EmbeddedLdapServer Tests

Issue gh-938
```